hbase-client/src/main/java/org/apache/hadoop/hbase/client/MasterRegistry.java

```diff
@@ -170,6 +214,11 @@ public static String getMasterAddr(Configuration conf) throws UnknownHostExcepti
     callable.call(controller, stub, resp -> {
       if (controller.failed()) {
         future.completeExceptionally(controller.getFailed());
+        // RPC has failed, trigger a refresh of master end points. We can have some spurious
```

Comment:
You sure that _any_ failure should require refreshing the list? A mechanism wherein RPC failure results in more RPCs has me suspicious...

Reply:
It seems like something we want to ensure is well-throttled, so that 100's of clients are not overwhelming 3 masters while they're trying to manage cluster recovery. I've recently been investigating an event where we have the rs hosting meta get overwhelmed by client requests, while the master is concurrently trying to process server crash procedures, thus the idea of the stampeding herd is fresh in my mind. Our IPC QoS story isn't great right now, so I want to think carefully about how failure and recovery scenarios play out with ratios of client:server that are on the order of 100:1.

Reply:
Ya, I think thats a valid concern. I was thinking about a similar edge case, hence the exponential back-off proposal. I think we can either implement a back-off or wrap it in a simple rate limiter (ex: no more than 1 getMasters in a 2 min window). WDYT?

Reply:
Between a throttle and an exponential backoff, i have no preference. I think we won't know what works until someone suffers the the consequences. Having something in place will address my concern.